### PR TITLE
[MS] Follow up fix to pass aligned args to variadic x86_32 functions

### PR DESCRIPTION
**Pull request by rnk**
MSVC allows users to pass structures with required alignments greater than 4 to variadic functions. It does not pass them indirectly to correctly align them. Instead, it passes them directly with the usual 4 byte stack alignment.

This change implements the same logic in clang on the passing side.  The receiving side (va_arg) never implemented any of this indirect logic, so it doesn't need to be updated.

This issue pre-existed, but @aaron.ballman noticed it when we started passing structs containing aligned fields indirectly in D152752.

### CODE REVIEW

**Review by AaronBallman — COMMENTED** [review]
I believe these changes are correct and they look to be reasonable, but adding the codegen code owners for final approval.

**Review by RKSimon — APPROVED** [review]
LGTM if the clang team are happy with us not raising the alignment error

**Comment by AaronBallman** [comment]
> LGTM if the clang team are happy with us not raising the alignment error

Yup, that's what got us on this path in the first place. Our downstream at Intel started getting errors on code that was previously working, this restores the previous behavior in that case.

**Review by AaronBallman — APPROVED** [review]
Approving, but do wait a bit before landing in case a codegen code owner has comments.

File: clang/lib/CodeGen/Targets/X86.cpp
Comment (rjmccall) [inline clang/lib/CodeGen/Targets/X86.cpp:824]: Minor request: please extract this second half into a helper function like `RequiredArgs::isRequiredArg(unsigned)`.

**Review by rjmccall — CHANGES_REQUESTED** [review]
Functionally LGTM; just a minor request.

File: clang/lib/CodeGen/Targets/X86.cpp
Comment (rnk) [reply clang/lib/CodeGen/Targets/X86.cpp]: Sure.

**Comment by AaronBallman** [comment]
Thank you for the fix, I appreciate it!

**Pull request merged**
